metal/cloud_test.go

```diff
@@ -211,14 +209,89 @@ func TestHasClusterID(t *testing.T) {
 }
 
 // builds an Equinix Metal client
-func constructClient(authToken string, baseURL *string) *packngo.Client {
-	client := retryablehttp.NewClient()
+func constructClient(authToken string, baseUrl *string) *metal.APIClient {
+	configuration := &metal.Configuration{
```

Comment:
Instead of setting all of these options here, you could do `configuration := metal.NewConfiguration()`, and then if `baseURL != nil` you can set `configuration.Servers` to point to `*baseURL`.